### DISCUSSION TITLE
2 주차 사전 과제, 우성현

### DISCUSSION THREAD

**Discussion by samuel426** [MrMirror21/2024-KHU-CSE-TrackStudy-Web#24] [General]
1. 배고픔이라는 상태가 있습니다.
해당 상태에서는 집중력이 낮아지고, 주변 반응에 예민해지며, 음식을 찾는 과정을 겪게 됩니다. 

2. 인프런
- 대쉬보드를 클릭시: 유저의 학습 현황을 한눈에 확인할 수 있는 ui 제공
- 학습목록 클릭시: 최근 학습순으로 정렬된 강의들을 확인 가능
- 목록 중 강의에 마우스를 올렸을 시: 항목별로 세부사항을 고를 수 있는 ui 제공

3.  결과: 1

이유:
x와 y는 같은 객체를 참조하고 있습니다.
let y = x;는 y에 x의 객체 참조를 할당합니다.
y.count = 1;는 해당 객체의 count 속성 값을 1로 변경합니다.
객체는 참조 타입이므로, y와 x는 같은 객체를 가리키기 때문에 x.count도 1로 변경됩니다.

4.
프로그래밍에서 불변성을 지키는 것은 코드의 안정성과 가독성을 높이는데 중요한 역할을 합니다.

5.
setTimeout은 비동기 함수로, 각각의 지연 시간 후에 콜백 함수를 실행하기에 결과는 다음과 같습니다.
시작
끝
1초 후
3초 후

**Comment by MrMirror21** [comment]
작성해주신 사전과제 잘 봤습니다. 전체적으로 잘 답변해주셨고, shallow copy, deep copy 문제나 JavaScript의 비동기 문제도 잘 풀어주신 것 같습니다. 성현님이 문제들에 어떤 답을 예측하셨는지 궁금했는데, 나중에 기회 되면 알려주시면 좋을 것 같습니다. 사전과제 작성이 지난 3주차 스터디 내용을 이해하는데 도움이 되셨으면 좋겠네요. 남은 스터디도 많은 참여 부탁드립니다 🙌
